classes/local/v1p2/container.php

```diff
@@ -0,0 +1,119 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * One Roster Enrolment Client.
+ *
+ * @package    enrol_oneroster
+ * @copyright  Andrew Nicols <andrew@nicols.co.uk>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace enrol_oneroster\local\v1p2;
+
+use enrol_oneroster\local\container as abstract_container;
+use enrol_oneroster\local\interfaces\cache_factory as cache_factory_interface;
+use enrol_oneroster\local\interfaces\collection_factory as collection_factory_interface;
+use enrol_oneroster\local\interfaces\container as container_interface;
+use enrol_oneroster\local\interfaces\entity_factory as entity_factory_interface;
+use enrol_oneroster\local\interfaces\filter as filter_interface;
+use enrol_oneroster\local\interfaces\rostering_endpoint as rostering_endpoint_interface;
+use enrol_oneroster\local\v1p2\factories\cache_factory;
+use enrol_oneroster\local\v1p2\factories\collection_factory;
+use enrol_oneroster\local\v1p2\factories\entity_factory;
+
+/**
+ * One Roster 1.2 Factory Manager.
+ *
+ * @package    enrol_oneroster
+ * @copyright  Andrew Nicols <andrew@nicols.co.uk>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class container extends abstract_container implements container_interface {
```

Comment:
Again these should all extend the v1p1 interface. That way ONLY the stuff that is different between p1 and p2 needs to be changed